detector/other/FieldMapBrBz.cpp

```diff
@@ -51,72 +71,60 @@ void FieldMapBrBz::fieldComponents(const double* pos , double* globalField) {
   const double posZ   = pos[2];
   const double posPhi = atan2(pos[1], pos[0]);
 
-  //Mokka defines these things as x and y and phi
-  double x = posRho;
-  double y = posZ;
+  double r   = posRho;
+  double z   = posZ;
   double phi = posPhi;
 
   //Get positive values to do less checks when comparing
-  if( y < 0 ) {
-    y = -y ;
-    phi += M_PI ;
+  if( z < 0 ) {
+    z   *= -1;
+    phi += M_PI;
   }
 
   //APS: Note the mokka field map does not start at 0, so we have to assume that
   //this area is covered, or add some more parameters for the values where the
   //field is supposed to be. Now we just assume it starts at 0/0/0, outside there is no field
-  if (not ( 0.0 <= x && x <= rhoMax &&
-	    0.0 <= y && y <= zMax ) ) {
+  if(r < rhoMin) r = rhoMin;
+  if(z < zMin  ) z = zMin;
+
+  if (not (r <= rhoMax && z <= zMax ) ) {
 
     return;
   }
 
-  if( x < rhoMin ) x = rhoMin;
-  if( y < zMin ) y = zMin ;
-
-
-  // Position of given point within region, normalized to the range
-  // [0,1]
-  const double xfraction = ( x - rhoMin ) / ( rhoMax - rhoMin );
-  const double yfraction = ( y - zMin )   / ( zMax   - zMin   );
+  int rBin,zBin;
+  double r0,z0;
+  double r1,z1;
 
-  // Need addresses of these to pass to modf below.
-  // modf uses its second argument as an OUTPUT argument.
-  double xdindex, ydindex;
+  rBin = int((r - rhoMin)/rhoStep);
+  zBin = int((z - zMin)/zStep);
```

Comment:
Now rBin and zBin are calculated just once. The getGlobalIndex function gets as inputs directly rBin and zBin.